design/cluster-update.md

```diff
@@ -0,0 +1,154 @@
+Of special note for removing storage is that a check should be performed to ensure that the cluster has enough remaining storage to recover (backfill) the entire set of objects from the OSD that is being removed.
+If the cluster does not have enough space for this (e.g., it would hit the `full` ratio), then the removal should not proceed.
+
+For each OSD to remove, the following steps should be performed:
+* mark the OSD as `out` with `ceph osd out <osd.id>`, which will trigger data migration from the OSD.
+* wait for all data to finish migrating from the OSD, meaning all placement groups return to the `active+clean` state
+* stop the OSD process and remove it from monitoring
+* remove the OSD from the CRUSH map: `ceph osd crush remove <osd.id>`
+* delete the OSD's auth info: `ceph auth del <osd.id>`
+* delete the OSD from the cluster: `ceph osd rm <osd.id>`
+* delete the OSD directory from local storage (if using `dataDirHostPath`): `rm -fr /var/lib/rook/<osd.id>`
+
+If the entire node is being removed, ensure that the host node is also removed from the CRUSH map:
+```console
+ceph osd crush rm <host-bucket-name>
+```
+
+#### Completion
+After all storage updates are completed, both additions and removals, then we can once again enable scrubbing:
+```console
+ceph osd unset noscrub
+ceph osd unset nodeep-scrub
+```
+
+### Placement Groups
+The number of placement groups in the cluster compared to the number of OSDs is a difficult trade-off without knowing the user's intent for future cluster growth.
+The general rule of thumb is that you want around 100 PGs per OSD.
+With less than that, you have potentially unbalanced distribution of data with certain OSDs storing more than others.
+With more PGs than that, you have increased overhead in the cluster because more OSDs need to coordinate with each other, impacting performance and reliability.
+
+It's important to note that **shrinking** placement group count (merging) is still **not supported** in Ceph.
+Therefore, you can only increase the number of placement groups (splitting) over time.
+
+If the cluster grows such that we have too few placement groups per OSD, then we can consider increasing the number of PGs in the cluster by incrementing the `pg_num` and `pgp_num` for each storage pool.
+Similar to adding new OSDs, this increase of PGs should be done incrementally and in a coordinated fashion to avoid degrading performance significantly in the cluster.
+
+Placement group management will be tracked in further detail in [issue 560](https://github.com/rook/rook/issues/560).
+
+## Scope
+The implementation of the design described in this document could be done in a phased approach in order to get critical features out sooner.
+One proposal for implementation phases would be:
+1. **Simple add/remove storage**: Storage resources can be added and removed from the Cluster CRD and extremely minimal orchestration would be performed to coordinate the storage changes.
```

Comment:
I would split add/remove phases

Reply:
That may be an acceptable split to make, delivering Add first since that is probably more useful to Rook users.